scheduledJobs/giveApples.js

```diff
@@ -0,0 +1,224 @@
+const cron = require('node-cron');
+const Promise = require('promise');
+const { fetchAllMessagesByChannelSince } = require('../common/discordutil');
+const { DateTime } = require('luxon');
+const { EmbedBuilder, MessageAttachment } = require('discord.js');
+
+const greenAppleRoleId = '875740793078431825';
+const redAppleRoleId = '875740253980336158';
+const rollCallRoleId = '763929191715831860';
+const ignoreReactionName = 'yoongerine';
+const greenAppleReactionName = 'green_apple';
+const logbookChannelNameStem = 'class-n-club-logbook';
+const jobDumpChannelId = '876672628692250654';
+
+module.exports = {
+    name: 'giveApples',
+    async start(client) {
+        // running job at 02:00 at America/Chicago timezone
+        cron.schedule('0 2 * * *', async () => {
+            const cst = 'America/Chicago';
+            const currentDateTimeCT = DateTime.utc().setZone(cst);
+            const messagesSinceDateTime = currentDateTimeCT.minus({ days: 1 });
+
+            const messagesWithApplesApplied = [];
+            let logbookUserIds = [];
+            let membersNeedingApples = [];
+            let membersNeedingGreenApples = [];
+            let usersWithGreenApple = '';
+            let usersWithRedApple = '';
+
+            const addAppleRoles = async () => {
+                const bothRoles = [greenAppleRoleId, redAppleRoleId];
+
+                const userFrequency = {};
+                for (let i = 0; i < logbookUserIds.length; i++) {
+                    userFrequency[logbookUserIds[i]] = userFrequency[logbookUserIds[i]] ? userFrequency[logbookUserIds[i]] + 1 : 1;
+                }
+
+                await Promise.all(membersNeedingApples.map(async (member) => {
```

Comment:
The `membersNeedingApples` saves all the members tagged, including duplicates. Maybe either filter them out before iterating through them, or don't save duplicates in `setMembersNeedingApples`
https://github.com/BangtanAcademyCoderCrew/Project-BangPD/blob/d0af3a7e4bfe0c5e39ceff420a7740956553fe41/scheduledJobs/giveApples.js#L93-L96

Maybe you could also count the frequency there instead of in addAppleRoles? 🤔

https://github.com/BangtanAcademyCoderCrew/Project-BangPD/blob/d0af3a7e4bfe0c5e39ceff420a7740956553fe41/scheduledJobs/giveApples.js#L31-L37

Reply:
going to use a Set to avoid duplicates. 👍